app/src/sharedTest/java/org/oppia/android/app/databinding/ColorBindingAdaptersTest.kt

```diff
@@ -0,0 +1,245 @@
+import org.oppia.android.testing.OppiaTestRule
+import org.oppia.android.testing.TestImageLoaderModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.firebase.TestAuthenticationModule
+import org.oppia.android.testing.junit.InitializeDefaultLocaleRule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestCoroutineDispatchers
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.AssetModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.EventLoggingConfigurationModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.SyncStatusModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.networking.NetworkConnectionDebugUtilModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [MarginBindingAdapters]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = ColorBindingAdaptersTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class ColorBindingAdaptersTest {
+  @get:Rule
+  val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+
+  @Inject
+  lateinit var context: Context
+
+  @get:Rule
+  val oppiaTestRule = OppiaTestRule()
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    Intents.init()
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  @Test
+  fun testColorBindingAdapters_setWhiteBackground_setsColorCorrectly() {
+    val whiteColor = context.getColor(R.color.component_color_shared_white_background_color)
+    launchActivity().use { scenario ->
+      scenario?.onActivity { activity ->
+        val testView: View = activity.findViewById(R.id.test_view)
+        setCustomBackgroundColor(testView, whiteColor)
+        val bg = testView.background as ColorDrawable
+        assertThat(getColorHexString(bg.color)).isEqualTo(getColorHexString(whiteColor))
+      }
+    }
+  }
+
+  @Test
+  fun testColorBindingAdapters_setBlackBackground_setsColorCorrectly() {
+    val blackColor = context.getColor(R.color.component_color_shared_black_background_color)
```

Comment:
For this & the above: I suggest just using a constant value to pass into both to ensure actual white/black colors are being used (vs. relying on the resource which might change over time and isn't actually obviously the color we expect just by looking at the test).

Reply:
Done